saleor/plugins/avatax/plugin.py

```diff
@@ -421,15 +424,20 @@ def _calculate_checkout_line_total_price(
             net = Decimal(line["lineAmount"])
 
             if currency == "JPY" and prices_entered_with_tax():
-                line_gross = base_value.gross
+                base_value._setup()
+                line_gross = Money(
+                    base_value._wrapped.amount - discount_amount, currency=currency
```

Comment:
Why call `._setup()` instead of using `base_value.amount` here and letting the simple object do its thing? If we need to deal with SimpleLazyObject directly, why do we even pass it as a lazy value?

Reply:
We are passing this as a lazy value, because we are using this `base_value` only for `JPY` currency or if there are no lines in response. So to avoid calling this method if it's not needed, the value is provided by `SimpleLazyObject`.

Reply:
But the whole point of `SimpleLazyObject` is that it's transparent. Why do we call `_setup()` and reach into the `_wrapped` object directly?